src/utils/scopes.ts

```diff
@@ -0,0 +1,13 @@
+import { OAuthScope } from "../entities/scope.entity.js";
+import { OAuthClient } from "../entities/client.entity.js";
+import { OAuthException } from "../exceptions/oauth.exception.js";
+
+export function guardAgainstInvalidClientScopes(client: OAuthClient, scopes: OAuthScope[]): void {
```

Comment:
this is a helper fn for users that throws if a requested scope is not valid on the client, I should add a doc for this